openwisp_controller/config/base/vpn.py

```diff
@@ -948,6 +966,52 @@ def _auto_ip(self):
                 return
         self.ip = self.vpn.subnet.request_ip()
 
+    def _generate_zt_identity(self):
+        try:
+            result = subprocess.run(
+                'zerotier-idtool generate',
+                shell=True,
+                check=True,
+                # in seconds
+                timeout=5,
+                capture_output=True,
+            )
+        except (CalledProcessError, TimeoutExpired) as exc:
+            err = getattr(exc, 'stderr', None)
+            # In case of timeout
+            if err is None:
+                err = exc
+            err_msg = f'Unable to generate zerotier identity secret, Error: {err}'
+            raise ZeroTierIdentityGenerationError(err_msg)
+        return result.stdout.decode('utf-8')
+
+    def _auto_zt_identity_secret(self):
+        if not self.zt_identity_secret and self.vpn._is_backend_type('zerotier'):
+            self.zt_identity_secret = self._generate_zt_identity()
+            self.member_id = self.zt_identity_secret[:10]
+
+    def _update_zt_network_member(self):
```

Comment:
Please move this method with other ZeroTier helper methods. We should group similar methods together for better readability.

Reply:
In my previous review, I got confused and thought that this method is part of the Vpn model. 

I think, this method should be a member of the Vpn model (Vpn._add_network_member). And here, we can perform `self.vpn._add_network_member(self.member_id)`.